src/actions/WalletActions.tsx

```diff
@@ -21,6 +26,104 @@ export interface SelectWalletTokenParams {
   alwaysActivate?: boolean
 }
 
+const activateWalletName: MapObject<{ name: string; notes: string }> = {
+  ripple: {
+    name: s.strings.activate_wallet_token_transaction_name_xrp,
+    notes: s.strings.activate_wallet_token_transaction_notes_xrp
+  }
+}
+
+const ACTIVATION_TOAST_AUTO_HIDE_MS = 5000
+
+const activateWalletTokens = async (
+  dispatch: Dispatch,
+  state: RootState,
+  navigation: NavigationBase,
+  wallet: EdgeCurrencyWallet,
+  tokenIds?: string[]
+): Promise<void> => {
+  if (tokenIds == null) throw new Error('Activating mainnet wallets unsupported')
+  const { account } = state.core
+  const { assetOptions } = await account.getActivationAssets({ activateWalletId: wallet.id, activateTokenIds: tokenIds })
+  const { pluginId } = wallet.currencyInfo
+  const { fiatCurrencyCode } = wallet
+
+  // See if there is only one wallet option for activation
+  if (assetOptions.length === 1 && assetOptions[0].paymentWalletId != null) {
```

Comment:
There is no `else` case on this guy. Should we at least put an `} else { // TODO }` block, or maybe show a warning?

Reply:
Sure. will do